### PR TITLE
atop: 2.8.1 -> 2.10.0

### PR DESCRIPTION
**Pull request by therishidesai**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/atop/default.nix [hunk 1]
```diff
@@ -12,20 +14,22 @@
 
 stdenv.mkDerivation rec {
   pname = "atop";
-  version = "2.8.1";
+  version = "2.10.0";
 
   src = fetchurl {
     url = "https://www.atoptool.nl/download/atop-${version}.tar.gz";
-    sha256 = "sha256-lwBYoZt5w0RPlx+FRXKg5jiR3C1fcDf/g3VwhUzg2h4=";
+    sha256 = "sha256-56ZzzyyCV4592C7LDeyD/Z7LMIKLJWHCip+lqvddX5M=";
```
Comment (jackyliu16) [inline pkgs/os-specific/linux/atop/default.nix:21]: Consider to replace sha256 with hash, currently is not recommanded to use sha256 as a param of fetcher.
```suggestion
    hash = "sha256-56ZzzyyCV4592C7LDeyD/Z7LMIKLJWHCip+lqvddX5M=";
```

File: pkgs/os-specific/linux/atop/atop.service.patch [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
 +++ b/atop.service
-@@ -9,5 +9,6 @@
- Environment=LOGPATH=/var/log/atop
+@@ -8,10 +8,11 @@ Environment="LOGOPTS="
+ Environment="LOGINTERVAL=600"
+ Environment="LOGGENERATIONS=28"
+ Environment="LOGPATH=/var/log/atop"
 -EnvironmentFile=/etc/default/atop
 +EnvironmentFile=-/etc/default/atop
+ ExecStartPre=/bin/sh -c 'test -d "${LOGPATH}" || mkdir -p "${LOGPATH}"'
```
Comment (7c6f434c) [inline pkgs/os-specific/linux/atop/atop.service.patch:11]: Doesn't `mkdir -p` silently succeed if the target exists already (so no need to test)? Line 14 in this patch suggests that the same directory is being ensured already.

Comment (therishidesai) [reply pkgs/os-specific/linux/atop/atop.service.patch:11]: Copied this from the previous patches for atop so not entirely sure the behavior here

Comment (7c6f434c) [reply pkgs/os-specific/linux/atop/atop.service.patch:11]: I don't think previous patches for `atop` contained any instance of `test -d`. And you are adding this line while the same functionality was already added by the same patch before your changes.

Comment (therishidesai) [reply pkgs/os-specific/linux/atop/atop.service.patch:11]: I think I got it from an older version. Will fix.

Comment (therishidesai) [reply pkgs/os-specific/linux/atop/atop.service.patch:11]: Oh actually this is in the upstream atop.service file that's why we see it in the patch file:

https://github.com/Atoptool/atop/blob/master/atop.service#L12

Comment (7c6f434c) [reply pkgs/os-specific/linux/atop/atop.service.patch:11]: Ah in this case we should probably drop the local addition of the same thing below, if we are changing the patch anyway

**Review by 7c6f434c — APPROVED** [review]
Looks good. Will obviously wait for CI.

**Pull request merged**